src/operator/nn/softmax.cc

```diff
@@ -187,5 +188,54 @@ NNVM_REGISTER_OP(_backward_softmax)
 #endif
 .set_attr<FCompute>("FCompute<cpu>", SoftmaxGradCompute<cpu, op::mshadow_op::mul,
                                                         mxnet_op::softmax_bwd>);
+
+NNVM_REGISTER_OP(masked_softmax)
+.add_alias("_npx_masked_softmax")
+.describe(R"code(Applies the softmax function masking elements according to the mask provided)code" ADD_FILELINE)
+.set_attr_parser(ParamParser<MaskedSoftmaxParam>)
+.set_attr<nnvm::FListOutputNames>("FListInputNames",
+    [](const NodeAttrs& attrs){
+    return std::vector<std::string>{"data", "mask"};
+})
+.set_attr<nnvm::FListOutputNames>("FListOutputNames",
+    [](const NodeAttrs& attrs) {
+    return std::vector<std::string>{"output"};
+})
+.set_attr<FCompute>("FCompute<cpu>", MaskedSoftmaxCompute<cpu, mxnet_op::softmax_fwd>)
+.set_attr<nnvm::FGradient>("FGradient", ElemwiseGradUseInOut{"_backward_masked_softmax"})
```

Comment:
We can try to register a `_zeros_like` for the mask.

See also https://github.com/apache/incubator-mxnet/blob/66488662ea6d33b8aeb64b632ea45194181f794a/src/operator/tensor/index_add_forward.cc#L113-L126

Reply:
is this not equivalent to https://github.com/apache/incubator-mxnet/blob/95a8d41bf48c7716df361c117aeef65fb120c79c/src/operator/nn/softmax-inl.h#L1642-L1646 
?

Reply:
I'm not sure but I guess in the graph-pass, the zeros_like nodes may potentially be optimized.